builtin-features/operations.inc

```diff
@@ -25,10 +25,39 @@ packToken Assign(const packToken& left, const packToken& right, evaluationData*
       // It is not possible to assign directly to
       // the global scope. It would be easy for the user
       // to do it by accident, thus:
-      if (!map || *map == TokenMap::default_global()) {
+      if (!map || *map == TokenMap::default_global()) 
+      {
         data->scope[var_name] = right;
-      } else {
-        (*map)[var_name] = right;
+      }
+      else 
+      {
+          packToken* assignee = map->find(var_name);
+
+          if (assignee)
```

Comment:
This check is redundant since assignee will always be present since the source map was returned from `.findMap()` and it is not null.

You could remove this if from around the `switch` below

Reply:
ok.